### PR TITLE
[keyboard] Add Windx65

### PR DESCRIPTION
**Pull request by thewerther**
This PR adds the QMK configuration for the windstudio windx65 keyboard and my personal keymap.

## Description

- Add files necessary for a working QMK configuration on the windx65
- personal keymap and configuration for reference

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/windx65/config.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 thewerther (@thewerther)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
```
Comment (waffle87) [inline keyboards/windx65/config.h:3]: This should all be moved to `info.json` file, as per the [PR checklist](https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs), and this file removed. You can use `qmk info -f json -kb windx65` to help with that.

File: keyboards/windx65/keymaps/thewerther/keymap.c [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            break;
+        case _DEFAULT_LAYER:
+            rgblight_sethsv_noeeprom(HSV_MAGENTA);
+            rgblight_mode_noeeprom(RGBLIGHT_MODE_RAINBOW_MOOD);
+            break;
+    }
+#   endif
+    return state;
+}
+
```
Comment (adophoxia) [inline keyboards/windx65/keymaps/thewerther/keymap.c:183]:
```suggestion

```
This should be in its own separate PR as a keymap PR if this board will be added.

File: keyboards/windx65/config.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#define RGB_DI_PIN D3
+#ifdef RGB_DI_PIN
+    #define RGBLED_NUM 6
+
+    #define RGBLIGHT_HUE_STEP 15
+    #define RGBLIGHT_SAT_STEP 15
+    #define RGBLIGHT_VAL_STEP 15
+    #define RGBLIGHT_SLEEP  /* If defined, the RGB lighting will be switched off when the host goes to sleep */
+    #define RGBLIGHT_ANIMATIONS
```
Comment (waffle87) [inline keyboards/windx65/config.h:34]: Animations should be enabled individually, not like this.

File: keyboards/windx65/info.json [hunk 4]
```diff
@@ -0,0 +1,353 @@
+{
+    "keyboard_name": "windx65",
+    "maintainer": "thewerther",
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (waffle87) [inline keyboards/windx65/info.json:5]:
```suggestion
        "LAYOUT": {
```

File: keyboards/windx65/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    ),
+};
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        writePin(D5, !led_state.caps_lock);
+    }
+    return res;
+}
```
Comment (waffle87) [inline keyboards/windx65/keymaps/default/keymap.c:43]:
```suggestion
```

File: keyboards/windx65/info.json [hunk 6]
```diff
@@ -0,0 +1,353 @@
+{
+    "keyboard_name": "windx65",
+    "maintainer": "thewerther",
```
Comment (waffle87) [inline keyboards/windx65/info.json:3]:
```suggestion
    "maintainer": "thewerther",
    "processor": "atmega32u4",
    "bootloader": "atmel-dfu",
    "features": {
        "bootmagic": true,
        "extrakey": true,
        "nkro": true,
        "rgblight": true
    },
    "indicators": {
        "caps_lock": "D5"
    },
```

File: keyboards/windx65/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    [0] = LAYOUT_all(
+        KC_ESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC,        KC_GRV,
+        KC_TAB,   KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC,              KC_DEL,
+        KC_CAPS,    KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_BSLS, KC_ENT, KC_NO,
+        KC_LSFT,      KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT,    KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT,            KC_SPC,                  MO(1), KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [1] = LAYOUT_all(
+        RESET,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  XXXXXXX,          XXXXXXX,
```
Comment (waffle87) [inline keyboards/windx65/keymaps/default/keymap.c:29]: This will have caused a compilation error. Please be testing your changes before blindly submitting them...
```suggestion
        QK_BOOT,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  XXXXXXX,          XXXXXXX,
```

File: keyboards/windx65/readme.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
+* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
+* **Keycode in layout**: Press the key mapped to `RESET` if it is available (FN + ESC)
```
Comment (waffle87) [inline keyboards/windx65/readme.md:25]:
```suggestion
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available (FN + ESC)
```

File: keyboards/windx65/rules.mk [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+RGBLIGHT_DRIVER = WS2812
```
Comment (waffle87) [inline keyboards/windx65/rules.mk:19]:
```suggestion
# This file intentionally blank
```

File: keyboards/windx65/windx65.h [hunk 10]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2022 thewerther (@thewerther)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
```
Comment (waffle87) [inline keyboards/windx65/windx65.h:3]: This should also be in `info.json`, and this file can be removed.

File: keyboards/windx65/windx65.c [hunk 11]
```diff
@@ -0,0 +1,3 @@
+// Copyright 2022 thewerther (@thewerther)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#include "windx65.h"
```
Comment (waffle87) [inline keyboards/windx65/windx65.c:3]: This file can be removed as well.

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by adophoxia — CHANGES_REQUESTED** [review]

File: keyboards/windx65/keymaps/thewerther/keymap.c (hunk 2 above)
Comment (waffle87) [reply keyboards/windx65/keymaps/thewerther/keymap.c:183]: > submitters can have a personal (or bells-and-whistles) keymap showcasing capabilities in the same PR but it shouldn’t be embedded in the ‘default’ keymap

This is fine to include.
